util/index.ts

```diff
@@ -0,0 +1,188 @@
+import { default as translateTimeline } from './translate_timeline';
+
+declare module 'inquirer' {
+  interface QuestionMap<T> {
+    fuzzypath: FuzzyPathQuestionOptions<T>;
+  }
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const getArgument = <T = string>(obj: any, propertyName: string): T | undefined => {
+  if (typeof obj === 'undefined' || obj === null)
+    return;
+
+  if (!(propertyName in obj))
+    return;
+
+  const prop = (obj as { [s: string]: unknown })[propertyName] as T;
+
+  if (typeof prop === 'undefined' || prop === null)
+    return;
+
+  return prop;
+};
+
+const dataFilesMap: { readonly [filename: string]: () => Promise<void> } = {
+  'effect_id': generateEffectIds,
+};
+
+const actionChoices = {
+  generate: {
+    name: 'Generate common data files',
+    value: 'generate',
+  },
+  translateTimeline: {
+    name: 'Translate Raidboss timeline',
+    value: 'translateTimeline',
+  },
+  findTranslations: {
+    name: 'Find missing translations',
+    value: 'findTranslations',
+  },
+};
+
+const argumentParser = new ArgumentParser({
+  description: 'A collection of common util functions for developing cactbot.',
+});
+const subparsers = argumentParser.addSubparsers({
+  title: 'action',
+  help: 'sub-command help',
+  dest: 'action',
+});
+
+const generateParser = subparsers.addParser(actionChoices.generate.value, {
+  description: actionChoices.generate.name,
+});
+
+generateParser.addArgument(['-c', '--choice'], {
+  choices: Object.keys(dataFilesMap),
+});
+
+const translateParser = subparsers.addParser(actionChoices.translateTimeline.value, {
+  description: actionChoices.translateTimeline.name,
+});
+
+translateParser.addArgument(['-l', '--locale'], {
+  type: 'string',
+  help: 'The locale to translate the timeline for, e.g. de',
+});
+translateParser.addArgument(['-t', '--timeline'], {
+  type: 'string',
+  help: 'The timeline file to match, e.g. "a12s"',
+});
+
+const findParser = subparsers.addParser(actionChoices.findTranslations.value, {
+  description: actionChoices.findTranslations.name,
+});
+
+findParser.addArgument(['-l', '--locale'], {
+  help: 'The locale to find missing translations for, e.g. de',
+});
+findParser.addArgument(['-f', '--filter'], {
+  nargs: '?',
+  type: 'string',
+  help: 'Limits the results to only match specific files/path',
+});
+
+inquirer.registerPrompt('fuzzypath', inquirerFuzzyPath);
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const run = (args: any) => {
+  inquirer.prompt([{
+    type: 'list',
+    name: 'action',
+    message: 'What do you want to do?',
+    choices: Object.values(actionChoices),
+    default: getArgument<string>(args, 'action'),
+    when: () => typeof getArgument(args, 'action') !== 'string',
+  }]).then((answer: Answers) => {
+    const action = getArgument(answer, 'action') || getArgument<string>(args, 'action');
+    if (action === actionChoices.generate.value)
+      return generateDataFiles(args);
```

Comment:
It should be fine to both `return` or not `return`, but I think it is better to let it knows that the function returns a `Promise` so that should be an async function?

Reply:
Ok!